### ISSUE TITLE
Spatial extension support

### ISSUE THREAD

**Issue by VGSML** [duckdb/pg_duckdb#141]
Do you plan to implement DuckDB spatial extension support, mainly ST_Read?
Will it be PostGIS type support?

**Comment by wuputah** [comment]
Yes, we would like to see spatial / GIS support, it's just a matter of time and priorities. Determining how it interacts with postgis will certainly need to be part of the design.

**Comment by VGSML** [comment]
Thanks, will wait.

**Comment by philiporlando** [comment]
+1 for PostGIS support!

**Comment by chong-spatial** [comment]
Does it support spatial extension yet?
looks like functions are not registered. I have run `postgres=# select duckdb.install_extension('spatial');` first. But when I run `postgres=# SELECT ST_AsText(ST_MakeEnvelope(0,0,1,1));`
ERROR:  function st_makeenvelope(integer, integer, integer, integer) does not exist
LINE 1: SELECT ST_AsText(ST_MakeEnvelope(0,0,1,1));
                         ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.

**Comment by VGSML** [comment]
I think it won't work like your example. The main difficulty to implement is postgis—duckdb function mapping. In the PG should be used normal Postgis functions or duckdb functions from the schema duckdb. Postgis geometry is stored in the ewkb format, duckdb stores wkb - without srid and BBBox information, than could add additional difficulty.
Also there is no geography type in the duckdb spatial extension.

A lot of things to do ....

**Comment by Maxxen** [comment]
> I think it won't work like your example. The main difficulty to implement is postgis—duckdb function mapping. In the PG should be used normal Postgis functions or duckdb functions from the schema duckdb. Postgis geometry is stored in the ewkb format, duckdb stores wkb - without srid and BBBox information, than could add additional difficulty. Also there is no geography type in the duckdb spatial extension.
> 
> A lot of things to do ....

PostGIS doesn't store EWKB internally, and DuckDB doesn't store WKB internally either. Both use their own internal format and while they are not identical, the DuckDB format is very much inspired by PostGIS (both store BBOX:es). The major difference is that DuckDB does not store the SRID in the geometry value itself.

I imagine the easiest solution is to just cast/convert to and from WKB when transferring geometries between DuckDB/PostGIS as ISO WKB basically models the common subset of spatial type support between the two systems.

**Comment by tjwebb** [comment]
+1
